eservice/lib/libpdo_enclave/contract_message.h

```diff
@@ -26,15 +26,15 @@ class ContractMessage
 {
 protected:
     bool VerifySignature(const ByteArray& signature) const;
+    void ComputeHash(ByteArray& message_hash) const;
 
 public:
     std::string expression_;
     std::string originator_verifying_key_;
     std::string channel_verifying_key_;
     std::string nonce_;
+    ByteArray message_hash_;
 
     ContractMessage(void){};
```

Comment:
OK, this comment does not belong here, not even related to this PR, but regards the usage of message hash, so going to write here anyway.

As of now, as part of a ccl update one stores the (S_i, S_i+1, messagehash) in the ledger. How is this information enough to recheck the execution? How does a second client get message corresponding to the messagehash

Reply:
it isn't sufficient. but the contract can/will store the message in the state. the hash of the message is provided to the verification function to ensure that the message registered on the ledger matches the most recent one.

Reply:
thank you! no sure I fully understand, but I hear that message is part of the state, so in the above example, message is part of S_i+1, that solves my problem.